columnflow/tasks/selection.py

```diff
@@ -354,27 +365,28 @@ def merge(self, inputs, output):
         law.pyarrow.merge_parquet_task(self, inputs, output)
 
     def zip_results_and_columns(self, inputs, tmp_dir):
-        import awkward as ak
         from columnflow.columnar_util import RouteFilter, sorted_ak_to_parquet
 
         chunks = []
 
         # setup the normalization weights producer
-        self.norm_weight_producer.run_setup(
-            self.requires()["forest_merge"]["normalization"],
-            self.input()["forest_merge"]["normalization"],
-        )
-
-        # define columns that will be written
-        write_columns = set(self.config_inst.x.keep_columns[self.task_family])
+        if self.dataset_inst.is_mc:
+            self.norm_weight_producer.run_setup(
+                self.requires()["forest_merge"]["normalization"],
+                self.input()["forest_merge"]["normalization"],
+            )
+
+        # define columns that will be written, including the event number as required for cutflows
+        write_columns = set(self.config_inst.x.keep_columns[self.task_family]) | {"event"}
```

Comment:
AFAIK event IDs are not unique, so usually one needs the full set of `run`, `luminosityBlock` and `event` to unambiguously identify an event. Should we maybe write out all three here?

Reply:
Yes, we should actually include all for unambiguous event book keeping during selection synchronizations. I added it and included a parameter for the choice of the counter variable.

Reply:
Nice! Then I guess this PR is good to go.